koku/api/iam/test/tests_serializers.py

```diff
@@ -219,11 +227,136 @@ def test_user_preference_duplicate(self):
 
         kwargs = {'context': {'user': user}}
         test_pref = {'foo': ['a', [1, 2, 3], {'b': 'c'}]}
-        data = {'name': 'foo', 'description': 'foo', 'preference': test_pref}
+        data = {'name': self.fake.word(),
+                'description': self.fake.text(),
+                'preference': test_pref}
         serializer = UserPreferenceSerializer(data=data, **kwargs)
         if serializer.is_valid(raise_exception=True):
             serializer.save()
 
+        with self.assertRaises(IntegrityError):
```

Comment:
The validation error was a fix for https://github.com/project-koku/koku/issues/309.

I'm pretty sure an Integrity Error will cause a 500 to display which isn't good.

Reply:
I'm open to suggestions on how to solve that. 

I _think_ the way to avoid `IntegrityError` is to do a lookup for the preference before setting it, so that we can pass the existing `UserPreference` in to the serializer, so it does an `update()` instead of a `create()`.

Reply:
Just a thought, but you could keep the logic they way it was, but add in an additional check to get the request METHOD. If its a POST throw the error, if its a PUT/PATCH don't throw the error.

Reply:
I've added a check in the view's `create()` method that will throw the proper error if the request is trying to create a preference that already exists.